### PR TITLE
Replace fetch with axios and fix slug-validation

### PR DESCRIPTION
**Pull request by norbye**
Replace fetch with axios to;
* shorten the amount of code required to perform API calls
* Enhance typing of the error responses
* Make a bigger chunk of the error response available to facilitate better error handling

Apparently django didn't do the magic validation I thought that it did in #1014 :'( so I fixed up a bit of that while I was at it (and added it to some older code that had frontend validation only + fixed up the tests to make it work). And a little bug with the admin-admission queryset not reloading until the backend reloaded.

Resolves ABA-459

### CODE REVIEW

File: frontend/src/routes/ManageAdmissions/CreateAdmission.tsx [hunk 1]
```diff
@@ -66,20 +67,22 @@ const CreateAdmission: React.FC = () => {
...
-              "Feilkode " + error?.code + ", klarte ikke lagre opptaket.",
-          });
-        } else {
-          setReturnedData({ type: "error", message: JSON.stringify(error) });
+        if (data?.slug) {
+          navigate("/admin/" + data.slug);
         }
       };
+      const onError = (error: AxiosError) => {
+        console.log(error.response?.data);
```
Comment (PeterJFB) [inline frontend/src/routes/ManageAdmissions/CreateAdmission.tsx:77]: `console.error` or yeet

File: frontend/src/query/mutations.ts [hunk 2]
```diff
@@ -18,13 +19,9 @@ interface CreateAdmissionProps {
...
 export const useAdminCreateAdmission = () => {
   const queryClient = useQueryClient();
-  return useMutation(
-    ({ admission }: CreateAdmissionProps) => {
-      return callApiFromQuery(`/admin/admission/`, {
-        method: "POST",
-        body: JSON.stringify(admission),
-      });
-    },
+  return useMutation<Record<string, unknown>, AxiosError, CreateAdmissionProps>(
```
Comment (PeterJFB) [inline frontend/src/query/mutations.ts:22]: Do we not know the type of `Record<string, unknown>`? :thinking:

**Review by PeterJFB — APPROVED** [review]
My django game is not strong rn, fix some small stuff then it looks good :100:

File: frontend/src/query/mutations.ts (hunk 2 above)
Comment (norbye) [reply frontend/src/query/mutations.ts:22]: Very good point, we do actually

**Pull request merged**